team-validator.js

```diff
@@ -253,16 +253,15 @@ class Validator {
 				if (!set.moves[i]) continue;
 				let move = tools.getMove(Tools.getString(set.moves[i]));
 				if (!move.exists) return ['"' + move.name + '" is an invalid move.'];
-				set.moves[i] = move.name;
```

Comment:
Here's one specific check that's failing: if you call `validateSet` twice on the same Keldeo-Resolute set, it will always fail the second time claiming that you need to have Secret Sword. This is because the check that the Pokemon ruleset uses is `set.moves.indexOf(toId(template.requiredMove)) < 0)` meaning that it relies on the moves in `set.moves` to be ids, not names. But because `validateSet` changes them to names, this doesn't work.
(Slightly inconveniently calling `validateSet` on invalid `Meloetta-Pirouette` or `Rayquaza-Mega` sets reset them to the base forme, which then passes validation the second time around. On the other hand this does at least ensure that Pikachu-Cosplay is changed to the appropriate forme, since the second time around the moves are names instead of ids and the test would fail.)
Maybe what I should be doing is calling `this.tools.getFormat('Pokemon').onChangeSet.call(this.tools, set, this.format);` instead of trying to call `validateSet` twice.